packages/financial-templates-lib/src/logger/PagerDutyV2Transport.ts

```diff
@@ -0,0 +1,52 @@
+// This transport enables winston logging to send messages to pager duty v2 api.
+import Transport from "winston-transport";
+import { event } from "@pagerduty/pdjs";
+
+type TransportOptions = ConstructorParameters<typeof Transport>[0];
+export type Severity = "critical" | "error" | "warning" | "info";
+
+interface Config {
+  integrationKey: string;
+  customServices?: { [key: string]: string };
+}
+
+export class PagerDutyV2Transport extends Transport {
+  private readonly integrationKey: string;
+  private readonly customServices: { [key: string]: string };
+  constructor(winstonOpts: TransportOptions, { integrationKey, customServices = {} }: Config) {
+    super(winstonOpts);
+    this.integrationKey = integrationKey;
+    this.customServices = customServices;
+  }
+  // pd v2 severity only supports critical, error, warning or info.
+  public static convertLevelToSeverity(level?: string): Severity {
+    if (!level) return "error";
+    if (level === "warn") return "warning";
+    if (level === "info" || level === "critical") return level;
+    return "error";
+  }
+  // Note: info must be any because that's what the base class uses.
+  async log(info: any, callback: () => void): Promise<void> {
+    try {
+      // we route to different pd services using the integration key (routing_key), or multiple services with the custom services object
+      const routing_key = this.customServices[info.notificationPath] ?? this.integrationKey;
+      await event({
+        data: {
+          routing_key,
+          event_action: "trigger",
+          payload: {
+            summary: `${info.level}: ${info.at} ⭢ ${info.message}`,
+            severity: PagerDutyV2Transport.convertLevelToSeverity(this.level),
+            source: info["bot-identifier"] ? info["bot-identifier"] : undefined,
+            // we can put any structured data in here as long as it is can be repped as json
+            custom_details: info,
```

Comment:
in the v1 bot we did some parsing on this to make it look different depending on if it has markdown in it or is a json. do you think it's worth doing the same here? https://github.com/UMAprotocol/protocol/blob/e4eb7213f1954afb2fc4d8d4e498562efedfcc1e/packages/financial-templates-lib/src/logger/PagerDutyTransport.ts#L38

Reply:
nvm I see that it spreads this nicely for us in this screenshot you posted: https://github.com/UMAprotocol/protocol/pull/3975#discussion_r963953415